### DISCUSSION TITLE
SleepMode

### DISCUSSION THREAD

**Discussion by o0shojo0o** [pixelit-project/PixelIt#25] [Node-RED Flows]
![image](https://user-images.githubusercontent.com/19333515/121034018-fbef5300-c7ac-11eb-8801-327dedbd6b4e.png)
Schaltet das Matrix Display dunkel und minimiert den Webservice der Pixel It.
Der Core sendet ebenfalls nichts mehr zur Pixel It :) 
~~~
[{"id":"3264e2bc.e3bcde","type":"function","z":"e3aa70fa.b265f","name":"SleepMode","func":"msg.sleepMode = false;\nnode.status({ fill: \"yellow\", shape: \"ring\", text: \"SleepMode is \" + msg.sleepMode});\nreturn msg;\n\n","outputs":1,"noerr":0,"x":230,"y":1180,"wires":[["869b90c9.21c74"]]},{"id":"44c27285.697d9c","type":"comment","z":"e3aa70fa.b265f","name":"SleepMode","info":"","x":90,"y":1140,"wires":[]},{"id":"869b90c9.21c74","type":"link out","z":"e3aa70fa.b265f","name":"SleepMode Automatic Link","links":["de85246c.e30708"],"x":335,"y":1180,"wires":[]},{"id":"ab00b1d3.fb1de","type":"inject","z":"e3aa70fa.b265f","name":"Init","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":"0","x":90,"y":1180,"wires":[["3264e2bc.e3bcde"]]}] 
~~~

**Comment by ojarosch** [comment]
Hi, ich habe eine Frage zum sleepMode, allerdings nicht in Verbindung mit Node-RED.
Ich habe aktuell die Beobachtung gemacht, dass ich nach einer Weile im sleepMode, in Timeouts laufe, wenn ich diesen per REST-Request wieder deaktivieren möchte.
Daher habe ich mich auf die Suche gemacht, was der sleepMode tatsächlich bewirkt.

Du schreibst:

> minimiert den Webservice der Pixel It

Im Code habe ich leider keine Referenz auf die Webservices gefunden, das widerspricht allerdings meinen Beobachtungen.
Habe ich etwas übersehen? Könntest du genauer beschreiben was der sleepMode bewirkt?

---

Über die API wird nur die Brightness auf 0 gesetzt
https://github.com/pixelit-project/PixelIt/blob/7309a6d91470ba4419da8fa52b7fe8a721ae99e5/src/PixelIt.ino#L1164-L1169

Über einen Buttonpress wird nach einer separaten Anzeige die Brightness auf 0 gesetzt.
https://github.com/pixelit-project/PixelIt/blob/7309a6d91470ba4419da8fa52b7fe8a721ae99e5/src/PixelIt.ino#L880-L888

**Reply by o0shojo0o** [reply]
Wie man hier sehen kann wird ein Teil der API nur bei sleepMode == false ausgewertet (der else Zweig)
https://github.com/pixelit-project/PixelIt/blob/7309a6d91470ba4419da8fa52b7fe8a721ae99e5/src/PixelIt.ino#L1165-L1171

**Reply by ojarosch** [reply]
Okey, damit kann ich nachvollziehen was du mit meintest. Das erklärt leider nicht warum mein ESP nicht auf den REST-Call antwortet.
Ich habe zuerst angenommen, dass der ESP einen tatsächlichen deepSleep Modus einnimmt und nur in kurzen Intervallen aufwacht, zeitweise die API anbieten oder nach Retained Messages auf dem MQTT-Broker sucht.
Code dazu habe ich in meiner kurzen Recherche nicht gefunden, daher die Frage.

**Reply by o0shojo0o** [reply]
> Okey, damit kann ich nachvollziehen was du mit meintest. Das erklärt leider nicht warum mein ESP nicht auf den REST-Call antwortet.

Ok das ist komisch, schon mal versucht den ESP(8266?) mir der [CleanFlash.bin](https://pixelit-project.github.io/firmware.html#firmware-2) zu flashen?

> Ich habe zuerst angenommen, dass der ESP einen tatsächlichen deepSleep Modus einnimmt und nur in kurzen Intervallen aufwacht, zeitweise die API anbieten oder nach Retained Messages auf dem MQTT-Broker sucht.

Ja das ist tatsächlich in der Dokumentation so gut umschrieben. Wenn Du da eine bessere Umschreibung hast, immer her damit! ([Hier kann man mitwirken](https://github.com/pixelit-project/Docs/blob/master/src/api.md#sleep-mode)) 

> Code dazu habe ich in meiner kurzen Recherche nicht gefunden, daher die Frage.

Kein Problem dafür sind wir ja hier im Discussions unterwegs.

**Reply by ojarosch** [reply]
> Wenn Du da eine bessere Umschreibung hast, immer her damit!

Siehe: https://github.com/pixelit-project/Docs/pull/1

> Ok das ist komisch, schon mal versucht den ESP(8266?) mir der [CleanFlash.bin](https://pixelit-project.github.io/firmware.html#firmware-2) zu flashen?

Tatsächlich noch nicht. Zwischenzeitlich hatte ich einen neuen ESP8266 (Wemos D1 mini) ausgepackt und geflasht, leider ohne anhaltenden Erfolg. Allerdings muss ich auch gestehen, dass ich eine Lösung auf anderem Wege gefunden habe: 
Auf dem Schreibtisch, wurde der `sleepMode` auf Anhieb deaktiviert. Im Regal nicht.
Ich tendiere dazu, dass der `sleepmode: false` Request von Home Assistant nicht ankam, da zuvor eine neue TCP-Verbindung aufgebaut werden musste. Node-RED hingegeben hält die Verbindung aufrecht.
Zusammen mit mehreren 2.4GHz Clienten über Wifi und BLE, hat der Verbindungsaufbau dann nicht geklappt.
Ein switch der die Node-RED Verbindung nutzt, funktioniert stattdessen tadellos.

**Reply by DanBendavid** [reply]
Hi all, 
The Sleep mode should trigger  to "unplug" the matrix as it draw +1W. We could use a IRFZ 44N  between GND_MATRIX and GND and D3 for HIGH or LOW.
What do you think ?
